can/interfaces/socketcan/socketcan.py

```diff
@@ -260,11 +321,24 @@ def build_bcm_update_header(can_id: int, msg_flags: int, nframes: int = 1) -> by
 
 
 def dissect_can_frame(frame: bytes) -> Tuple[int, int, int, bytes]:
-    can_id, can_dlc, flags = CAN_FRAME_HEADER_STRUCT.unpack_from(frame)
+    can_id, data_len, flags, len8_dlc = CAN_FRAME_HEADER_STRUCT.unpack_from(frame)
     if len(frame) != constants.CANFD_MTU:
         # Flags not valid in non-FD frames
         flags = 0
-    return can_id, can_dlc, flags, frame[8 : 8 + can_dlc]
+
+    if data_len not in can.util.CAN_FD_DLC:
+        data_len = min(i for i in can.util.CAN_FD_DLC if i >= data_len)
+
+    # Allow deprecated can frames with old struct
+    if (
+        data_len == constants.CAN_MAX_DLEN
+        and constants.CAN_MAX_DLEN < len8_dlc <= constants.CAN_MAX_RAW_DLC
+    ):
+        can_dlc = len8_dlc
+    else:
+        can_dlc = data_len
+
+    return can_id, can_dlc, flags, frame[8 : 8 + data_len]
```

Comment:
This is very nit-picky: I know this wasn't any better before the new PR, but I don't like that we don't check that the expected `data_len` is actually present in the frame. Since it's Python it will simply return less data than expected.

Reply:
Sure I can add this verification to make it more robust. Should I add padding bytes , recude the data_len value or raise an exception in that case? Not sure what will be the best here...

Reply:
@lumagi - what approach do you think will be better?

Reply:
I didn't think about the fact that the buffer returned from the kernel always has the maximum length and so the slicing operation should always provide the necessary amount of bytes. Please disregard my comment.